tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp

```diff
@@ -1223,26 +1223,45 @@ ASTBuildOperationRef ASTProducer::getBuildOperationForConsumer(
     SwiftASTConsumerRef Consumer,
     IntrusiveRefCntPtr<llvm::vfs::FileSystem> FileSystem,
     SwiftASTManagerRef Mgr) {
+  ASTBuildOperationRef LatestMatch;
```

Comment:
I would add a comment that explains which operation is expected to be returned 

```cpp
/// If there is a finished build operation that 'Consumer' can use (based on 'canUseASTWithSnapshots') return that. If no such operation exists and a in-progress operation exists that exactly matches the source state of the consumer, return that. If neither a finished operation that can be used nor an in-progress operation with the exact same source state exists, return `nullptr`.
```

Reply:
This isn't what's returned at the moment. Unless I messed this up it should be:
  - Prefer any complete build operation that exactly matches or can be used with snapshots
  - If none, use the latest in-progress build operation with the same condition
  - Otherwise return nullptr

Reply:
Ah, yes, I think you’re right. That shows what the doc comment is good for 😉